common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -125,15 +125,17 @@ public ShuffleIndexInformation load(String filePath) throws IOException {
       .weigher((Weigher<String, ShuffleIndexInformation>)
         (filePath, indexInfo) -> indexInfo.getRetainedMemorySize())
       .build(indexCacheLoader);
-    DBBackend dbBackend = null;
+    DBBackend dbBackend;
     if (registeredExecutorFile != null) {
```

Comment:
Do we really need this condition? I feel like it's useless.

Reply:
hmm... `registeredExecutorFile` may be null, for example, the following code path:

https://github.com/apache/spark/blob/68c47d5e74b8be481318388b3cc8b40ead35beea/core/src/main/scala/org/apache/spark/deploy/ExternalShuffleService.scala#L81-L92

https://github.com/apache/spark/blob/68c47d5e74b8be481318388b3cc8b40ead35beea/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java#L79-L84

However, without adding this condition, the result will not change

Reply:
And the log will look strange without this condition

Reply:
[e3500cf](https://github.com/apache/spark/pull/37648/commits/e3500cf714f9789604c1b977a5c64b82bd9337bb) remove the condition

Reply:
Wait CI